### PR TITLE
Use atparse for variable subsitution in MOM and CICE input files

### PR DESCRIPTION
**Pull request by DeniseWorthen**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

Removes use of ``edit_inputs`` for MOM input templates and the ice_in template and uses the ``atparse`` function in ``run_test`` instead.

### Issue(s) addressed

Issue #783 

## Testing

Initial testing on Cheyenne.intel for all coupled tests passed. All non-standalone tests on hera.intel pass.

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [ ] hera.intel
- [ ] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel 
- [ ] cheyenne.gnu
- [ ] gaea.intel 
- [ ] jet.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3
- [ ] CI

## Dependencies

None

### CODE REVIEW

File: tests/parm/ice_in_template [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
     ndtd           = 1
-    runtype        = 'CICERUNTYPE'
-    runid          = 'RUNID'
+    runtype        = '@[CICERUNTYPE]'
+    runid          = '@[RUNID]'
     ice_ic         = 'cice_model.res.nc'
     restart        = .true.
-    restart_ext    = RESTART_EXT
-    use_restart_time = USE_RESTART_TIME
+    restart_ext    = .false.
```
Comment (aerorahul) [inline tests/parm/ice_in_template:14]: Shouldn't this be:
```suggestion
    restart_ext    = @[RESTART_EXT]
```

**Review by aerorahul — COMMENTED** [review]
Changes are in line with the description and seem correct to the visual review.
I found one place where perhaps the changes did not translate correctly, or were intentional.
Please take a look at the comment.
I would be happy to approve after your response to the comment.

File: tests/parm/ice_in_template (hunk 1 above)
Comment (DeniseWorthen) [reply tests/parm/ice_in_template:14]: I decided to remove this as a configurable option since we now have it set false by default always.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by aerorahul — APPROVED** [review]
Thanks for the comment.

In the future, if the GODAS application needs this option to be configurable, we can address it then.

**Comment by aerorahul** [comment]
@DeniseWorthen 
I confirmed with NG-GODAS that currently `restart_ext = .false.` in their  application.  
However, they wondered if you would be open in keeping this as a configurable option?

**Comment by DeniseWorthen** [comment]
We have not used this variable in a non-default (.false.) setting at least since updating to CICE6. Yes, we could keep it as a configurable variable, but I do not believe it serves a purpose to do so.

**Comment by climbfuji** [comment]
Note that I already pulled this into my PR #792 - no need to update from develop.

**Comment by DeniseWorthen** [comment]
Will be committed as part of PR #792; closing.

**Comment by DeniseWorthen** [comment]
Changes will be committed in PR #791; Closing.

**Pull request closed without merging**